src/main/java/lembas/core/LembasCore.java

```diff
@@ -0,0 +1,126 @@
+import lembas.core.api.server.LembasCoreServer;
+import lembas.core.internal.client.LembasCoreClientImpl;
+import lembas.core.internal.common.LembasCoreCommonImpl;
+import lembas.core.internal.server.LembasCoreServerImpl;
+
+/**
+ * The main mod file of Lembas Core. Everything which can be accessed from the
+ * outside (modid, version, getClient(), ...) is considered to be a part of the
+ * API. Internal implementation details (for example the proxy implementation
+ * classes) ARE NOT PART OF THE API.
+ *
+ */
+@Mod(modid = LembasCore.MODID, version = LembasCore.VERSION, name = "Lembas Core", acceptedMinecraftVersions = "*")
+public class LembasCore
+{
+
+    public static final String MODID = "lembas-core";
+    public static final String VERSION = "@version@"; // Injected by gradle
+
+    @Instance(value = LembasCore.MODID)
+    public static LembasCore instance;
+
+    @SidedProxy(clientSide = "lembas.core.internal.client.LembasCoreClientImpl", serverSide = "lembas.core.internal.server.LembasCoreServerImplF", modId = LembasCore.MODID)
+    public static LembasCoreCommonImpl proxy;
+
+    private LembasCoreClientImpl client;
+    private LembasCoreServerImpl server;
+
+    /**
+     * Returns the client proxy instance. If called server-side, an exception will
+     * be thrown. The instance won't change - it can be cached by the caller.
+     * 
+     * @return The client proxy instance
+     */
+    public LembasCoreClient getClient ()
+    {
+        if (client == null)
+        {
+            client = (LembasCoreClientImpl) proxy;
+        }
+        return client;
+    }
+
+    /**
+     * Returns the server proxy instance - only if called on the PHYSICAL server. If
+     * called client-side or via an integrated server, an exception will be thrown.
+     * The instance won't change - it can be cached by the caller.
+     * 
+     * @return The server proxy instance
+     */
+    public LembasCoreServer getServer ()
+    {
+        if (server == null)
+        {
+            server = (LembasCoreServerImpl) proxy;
+        }
+        return server;
+    }
+
+    @EventHandler
+    public void onPreInit (FMLPreInitializationEvent event)
+    {
+        proxy.onPreInit (event);
+    }
+
+    @EventHandler
+    public void onInit (FMLInitializationEvent event)
+    {
+        proxy.onInit (event);
+    }
+
+    @EventHandler
+    public void onPostInit (FMLPostInitializationEvent event)
+    {
+        proxy.onPostInit (event);
+    }
+
+    @EventHandler
+    public void onLoadingComplete (FMLLoadCompleteEvent event)
+    {
+        proxy.onLoadingComplete (event);
+    }
+
+    @EventHandler
+    public void onServerStarting (FMLServerStartingEvent event)
+    {
+        // Will be executed for integrated servers too - but the proxy is not the server
+        // proxy there
+        if (proxy instanceof LembasCoreServer)
+        {
+            ((LembasCoreServerImpl) proxy).onServerStarting (event);

```

Comment:
You are testing for an instance of `LembasCoreServer` and then immediately cast it to `LembasCoreServerImpl`. Though there should not be a scenario where this causes an error, I think it is safer and better practice to change the instanceof test to `LembasCoreServerImpl`.

Reply:
Yeah, you're right.

Reply:
Changed to 9baaa6a5d8758795a8bbce829771c86691da0cae